numpy/__init__.pyi

```diff
@@ -2169,6 +2169,7 @@ class ndarray(_ArrayOrScalarCommon, Generic[_ShapeT_co, _DTypeT_co]):
     def ctypes(self) -> _ctypes[int]: ...
     @property
     def shape(self) -> _ShapeT_co: ...
+    @deprecated("Setting the shape on a NumPy array has been deprecated in NumPy 2.4")
     @shape.setter
```

Comment:
I'm not sure why, but my instinct tells me that these two should be swapped

```suggestion
    @shape.setter
    @deprecated("Setting the shape on a NumPy array has been deprecated in NumPy 2.4")
```

That's probably because I know from experience that this way it'll work for both mypy and pyright, whereas I'm not sure about the other way around. But then again, maybe it doesn't matter 🤷🏻

Reply:
I followed the style a couple of lines below for `strides.setter`. Could you check how this works with mypy and pyright?

Reply:
mypy indeed doesn't seem to like it this way:

<img width="699" height="124" alt="image" src="https://github.com/user-attachments/assets/8a4787da-8610-49cc-93ad-b2755ec1765d" />

Swapping them seems to help:

<img width="697" height="97" alt="image" src="https://github.com/user-attachments/assets/b896c759-5ad2-4c51-826b-b4c7b8069e6c" />